src/hybrid_reader.c

```diff
@@ -403,6 +403,10 @@ void HybridIterator_Free(struct indexIterator *self) {
 }
 
 IndexIterator *NewHybridVectorIterator(HybridIteratorParams hParams) {
+  RS_LOG_ASSERT(
+      hParams.qParams.searchMode >= 0 && hParams.qParams.searchMode < VECSIM_LAST_SEARCHMODE,
+      "Invalid hybrid search mode");
+
```

Comment:
This will send us to hell. Aren't we over-reacting?

Reply:
It will protect us from getting valid but wrong results. 
Also, users should not call this function directly so it can only guarantee we are not adding any bugs before releasing new code.